mapiproxy/libmapistore/backends/indexing_mysql.c

```diff
@@ -233,65 +589,16 @@ static enum mapistore_error mysql_record_del(struct indexing_context *ictx,
 		return MAPISTORE_ERR_INVALID_PARAMETER;
 	}
 
+	retval = mysql_record_get_uri(ictx, username, mem_ctx, fmid, &uri, &IsSoftDeleted);
+	MAPISTORE_RETVAL_IF(retval, retval, mem_ctx);
```

Comment:
It is required but we are not using the function to check if the record exists but to retrieve the uri we  are next going to delete from memcached cache.

Otherwise we end up with an execute_query that will have removed the uri when MAPISTORE_PERMANENT_DELETE is set and prevents from also removing it from cache.